Quantum-Control-Applications/Superconducting/Single-Fixed-Transmon/04b_resonator_spectroscopy_wide_range_octave.py

```diff
@@ -0,0 +1,175 @@
+    f = declare(int)  # QUA variable for the resonator IF
+    I = declare(fixed)  # QUA variable for the measured 'I' quadrature
+    Q = declare(fixed)  # QUA variable for the measured 'Q' quadrature
+    I_st = declare_stream()  # Stream for the 'I' quadrature
+    Q_st = declare_stream()  # Stream for the 'Q' quadrature
+    n_st = declare_stream()  # Stream for the averaging iteration 'n'
+
+    with for_(i, 0, i < len(LOs) + 1, i + 1):
+        pause()  # This waits until it is resumed from python
+        with for_(n, 0, n < n_avg, n + 1):
+            with for_(*from_array(f, IFs)):
+                # Update the frequency of the digital oscillator linked to the qubit element
+                update_frequency("resonator", f)
+                # Measure the state of the resonator
+                measure(
+                    "readout",
+                    "resonator",
+                    None,
+                    dual_demod.full("cos", "out1", "sin", "out2", I),
+                    dual_demod.full("minus_sin", "out1", "cos", "out2", Q),
+                )
+                # Wait for the resonator to deplete
+                wait(depletion_time * u.ns, "resonator")
+                # Save the 'I' & 'Q' quadratures to their respective streams
+                save(I, I_st)
+                save(Q, Q_st)
+        # Save the LO iteration to get the progress bar
+        save(i, n_st)
+
+    with stream_processing():
+        # Cast the data into a 2D matrix, average the matrix along its second dimension (of size 'n_avg') and store the results
+        # (1D vector) on the OPX processor
+        I_st.buffer(len(IFs)).buffer(n_avg).map(FUNCTIONS.average()).save_all("I")
+        Q_st.buffer(len(IFs)).buffer(n_avg).map(FUNCTIONS.average()).save_all("Q")
+        n_st.save_all("iteration")
+
+#####################################
+#  Open Communication with the QOP  #
+#####################################
+qmm = QuantumMachinesManager(host=qop_ip, port=qop_port, cluster_name=cluster_name, octave=octave_config)
+
+
+###############
+# Run Program #
+###############
+# Open the quantum machine
+qm = qmm.open_qm(config)
+
+# Calibrate the element for each LO frequency of the sweep and the central intermediate frequency
+calibrate = True
+if calibrate:
+    for lo in LOs:
+        print(f"Calibrate (LO, IF) = ({lo/u.MHz}, {IFs[len(IFs) // 2]/u.MHz}) MHz")
+        qm.calibrate_element("resonator", {lo: (IFs[len(IFs) // 2],)})
+
+# Send the QUA program to the OPX, which compiles and executes it. It will stop at the 'pause' statement.
+job = qm.execute(resonator_spec)
+# Creates results handles to fetch the data
+res_handles = job.result_handles
+# Initialize empty vectors to store the global 'I' & 'Q' results
+I_tot = []
+Q_tot = []
+# Live plotting
+fig = plt.figure()
+interrupt_on_close(fig, job)  # Interrupts the job when closing the figure
+
+for i in range(len(LOs)):  # Loop over the LO frequencies
```

Comment:
Maybe a matter of personal preference but I prefere to use `enumerate` instead of `range(len[])`


```suggestion
for ii, LO in enumerate(LOs):  # Loop over the LO frequencies
```